client/src/components/SearchBox.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import React from 'react';
+import Form from 'react-bootstrap/Form';
+import Col from 'react-bootstrap/Col';
+
+type Props = {
+  placeholderText: string
+};
+
+type State = {
+  value: string
+};
+
+export class SearchBox extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {value: ''};
+    this.handleChange = this.handleChange.bind(this);
+  }
+  timer: TimeoutID;
+
+  /*:: handleChange: () => void */
+  handleChange(e: SyntheticInputEvent<*>) {
+    this.setState(
+      {value: e.target.value},
+      () => {
+        if (this.timer) clearTimeout(this.timer);
+        if (this.state.value.length !== 1) {
```

Comment:
important: never use `this.state` in `setState`,
see [here](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) why

Reply:
That's the only way I could make this function work.
If you know another way, I don't mind

Reply:
you can refer current `state` inside `setState`, BUT, since the update is async, you should pass state to setState instead of using `this.state`.

e.g:
🔴wrong:
```js
this.setState({
  counter: this.state.counter + 1,
});
```
✅correct:
```js
this.setState((state) => ({
  counter: state.counter + 1
}));
```

Reply:
I saw this part too, but it's not working for me.

Reply:
my mistake, you use it in the callback part and that's fine 👍🏽